Subject: [PATCH Isabelle2011-1] tracing command terms and rule instantiations (requires src/Pure/Tools/pproc.ML)

diff -rup src/Pure/IsaMakefile src-patched/Pure/IsaMakefile
--- src/Pure/IsaMakefile	2011-10-09 15:47:58.000000000 +0100
+++ src-patched/Pure/IsaMakefile	2012-05-31 09:03:11.000000000 +0100
@@ -209,6 +209,7 @@ $(OUT)/Pure: $(BOOTSTRAP_FILES)				\
   Tools/find_consts.ML					\
   Tools/find_theorems.ML				\
   Tools/named_thms.ML					\
+  Tools/pproc.ML					\
   Tools/xml_syntax.ML					\
   assumption.ML						\
   axclass.ML						\
diff -rup src/Pure/Isar/rule_insts.ML src-patched/Pure/Isar/rule_insts.ML
--- src/Pure/Isar/rule_insts.ML	2011-10-09 15:47:58.000000000 +0100
+++ src-patched/Pure/Isar/rule_insts.ML	2012-05-31 10:42:01.000000000 +0100
@@ -87,7 +87,7 @@ fun read_termTs ctxt schematic ss Ts =
       |> Variable.polymorphic ctxt;
     val Ts' = map Term.fastype_of ts';
     val tyenv = fold Type.raw_match (Ts ~~ Ts') Vartab.empty;
-  in (ts', map (apsnd snd) (Vartab.dest tyenv)) end;
+  in (PProc.trace_cmd_terms ss ts'; (ts', map (apsnd snd) (Vartab.dest tyenv))) end;
 
 fun read_insts ctxt mixed_insts (tvars, vars) =
   let
@@ -173,6 +173,7 @@ fun read_instantiate_mixed ctxt mixed_in
         else
           Token.assign (SOME (Token.Term (the (AList.lookup (op =) term_insts xi)))) arg);
   in
+    PProc.trace_inst_names mixed_insts;
     Drule.instantiate_normalize insts thm |> Rule_Cases.save thm
   end;
 
diff -rup src/Pure/ROOT.ML src-patched/Pure/ROOT.ML
--- src/Pure/ROOT.ML	2011-10-09 16:13:41.000000000 +0100
+++ src-patched/Pure/ROOT.ML	2012-05-31 14:46:31.000000000 +0100
@@ -253,6 +253,9 @@ use "PIDE/document.ML";
 use "Thy/rail.ML";
 
 (*theory and proof operations*)
+
+use "Tools/pproc.ML";
+
 use "Isar/rule_insts.ML";
 use "Thy/thm_deps.ML";
 use "Isar/isar_cmd.ML";
